### PR TITLE
initdb cleanup and alignment with upstream

### PR DESCRIPTION
**Pull request by danielgustafsson**
Contained are a set of cleanup commits to initdb which further aligns initdb with upstream by removing unused (or not needed) Greenplum extensions, and adds back some pieces of upstream. See each commit message for more information.

One open question is how to handle the bgwriter section from the sample config, it's completely missing in Greenplum and the question is if that was intentional or an oversight.

### CODE REVIEW

File: src/bin/initdb/initdb.c [hunk 1]
```diff
@@ -3740,10 +3721,10 @@ main(int argc, char *argv[])
 				xlog_dir = pg_strdup(optarg);
 				break;
 			case 1001:
-                n_connections = parse_long(optarg, false, optname);
+				n_connections = parse_long(optarg, false, "max_connection");
```
Comment (jchampio) [inline src/bin/initdb/initdb.c]: `long_options[option_index].name` should work too, I think. The removed code section tries to put a lot of ceremony around it but I think the author probably misunderstood when the long option index was valid.

Or, if you find the hardcoding more clear...

File: src/bin/initdb/initdb.c [hunk 2]
```diff
@@ -3631,25 +3631,6 @@ main(int argc, char *argv[])
 
 	while ((c = getopt_long(argc, argv, "dD:E:kL:nNU:WA:sST:X:", long_options, &option_index)) != -1)
```
Comment (jchampio) [inline src/bin/initdb/initdb.c]: ...then I think `option_index` is now unused and can go away.

Comment (danielgustafsson) [reply src/bin/initdb/initdb.c]: Nice catch, fixed.

File: src/bin/initdb/initdb.c (hunk 1 above)
Comment (danielgustafsson) [reply src/bin/initdb/initdb.c]: Yeah, I don't think the added complexity is worth it for just two callsites.

**Comment by danielgustafsson** [comment]
Pushed to master with a small gcc-8 compilation fix (which will be completely taken care of in the current merge).

**Pull request closed without merging**